test/cbmc/proofs/MQTTAgentCommand_Connect/MQTTAgentCommand_Connect_harness.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * coreMQTT-Agent v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file MQTTAgentCommand_Connect_harness.c
+ * @brief Implements the proof harness for MQTTAgentCommand_Connect function.
+ */
+
+/* MQTT agent include. */
+#include "mqtt_agent_command_functions.h"
+
+void harness()
+{
+    MQTTAgentContext_t * pMqttAgentContext;
+    MQTTAgentCommandFuncReturns_t * pReturnFlags;
+    MQTTAgentConnectArgs_t * pConnectArgs;
+
+    pMqttAgentContext = malloc( sizeof( MQTTAgentContext_t ) );
+    __CPROVER_assume( pMqttAgentContext != NULL );
+    pReturnFlags = malloc( sizeof( MQTTAgentCommandFuncReturns_t ) );
+    __CPROVER_assume( pReturnFlags != NULL );
+    pConnectArgs = malloc( sizeof( MQTTAgentConnectArgs_t ) );
```

Comment:
Same here. The purpose of CBMC is to define properties with `__CPROVER_assume` about input parameters given to the function under test. By simply calling `malloc`, the only property we are stating here is that `pConnectArgs` contains memory on the heap with the same size as `MQTTAgentConnectArgs_t`. That means that each member of the struct can take on any value, but do we really want that to be the case? Sure, `sessionPresent` can take on 'true' or 'false'. The timeout can also take on any value. However, do we expect `MQTTAgentCommand_Connect` to be called with `pConnectInfo` pointing to any random address? No, we only expect  `MQTTAgentCommand_Connect` to be called when `pConnectInfo` points to a `MQTTConnectInfo_t` structure.

In summary, you need to create an `allocateConnectArgs( MQTTAgentConnectArgs_t * pConnectArgs )` function that will set `MQTTAgentConnectArgs_t` to values that we expect its caller (`MQTTAgentCommand_Connect`) to see. This applies to everything we do in CBMC.